internal/controller/config.go

```diff
@@ -22,67 +22,122 @@ import (
 	LeaderElectionResourceLock string
 }
 
 // ManagerConfig defines the configuration for the controller manager.
 type ManagerConfig struct {
 	// MetricsAddr is the address the metric endpoint binds to.
+	// Deprecated: This field will be eventually removed. Please use Server.Metrics.BindAddress instead.
 	MetricsAddr string
+	// Server is the configuration for the HTTP server.
+	Server *ServerConfig
 	LeaderElectionConfig
 	// DisableLeaseCache specifies whether to disable cache for lease.coordination.k8s.io resources.
 	DisableLeaseCache bool
-	// IgnoreOperationAnnotation specifies whether to ignore or honour the operation annotation on resources to be reconciled.
-	IgnoreOperationAnnotation bool
 	// FeatureGates contains the feature gates to be used by etcd-druid.
 	FeatureGates featuregate.MutableFeatureGate
 	// EtcdControllerConfig is the configuration required for etcd controller.
 	EtcdControllerConfig *etcd.Config
-	// CustodianControllerConfig is the configuration required for custodian controller.
-	CustodianControllerConfig *custodian.Config
 	// CompactionControllerConfig is the configuration required for compaction controller.
 	CompactionControllerConfig *compaction.Config
 	// EtcdCopyBackupsTaskControllerConfig is the configuration required for etcd-copy-backup-tasks controller.
 	EtcdCopyBackupsTaskControllerConfig *etcdcopybackupstask.Config
 	// SecretControllerConfig is the configuration required for secret controller.
 	SecretControllerConfig *secret.Config
+	// SentinelWebhookConfig is the configuration required for sentinel webhook.
+	SentinelWebhookConfig *sentinel.Config
 }
 
 // InitFromFlags initializes the controller manager config from the provided CLI flag set.
 func (cfg *ManagerConfig) InitFromFlags(fs *flag.FlagSet) error {
-	flag.StringVar(&cfg.MetricsAddr, metricsAddrFlagName, defaultMetricsAddr, ""+
-		"The address the metric endpoint binds to.")
+	cfg.Server = &ServerConfig{}
+	cfg.Server.Metrics = &Server{}
+	cfg.Server.Webhook = HTTPSServer{}
+	cfg.Server.Webhook.Server = Server{}
+
+	flag.StringVar(&cfg.Server.Metrics.BindAddress, metricsBindAddressFlagName, defaultMetricsBindAddress,
+		"The IP address that the metrics endpoint binds to.")
+	flag.IntVar(&cfg.Server.Metrics.Port, metricsPortFlagName, defaultMetricsPort,
+		"The port used for the metrics endpoint.")
+	flag.StringVar(&cfg.Server.Metrics.BindAddress, metricsAddrFlagName, defaultMetricsAddr,
+		fmt.Sprintf("The fully qualified address:port that the metrics endpoint binds to. Deprecated: this field will be eventually removed. Please use %s and %s instead.", metricsBindAddressFlagName, metricsPortFlagName))
+	flag.StringVar(&cfg.Server.Webhook.Server.BindAddress, webhookServerBindAddressFlagName, defaultWebhookServerBindAddress,
+		"The IP address on which to listen for the HTTPS webhook server.")
+	flag.IntVar(&cfg.Server.Webhook.Server.Port, webhookServerPortFlagName, defaultWebhookServerPort,
```

Comment:
https://github.com/gardener/etcd-druid/pull/777#discussion_r1589992549
```suggestion
	flag.Uint16Var(&cfg.Server.Webhook.Server.Port, webhookServerPortFlagName, defaultWebhookServerPort,
```

Reply:
already changed.